air/src/execution_step/errors/intructions_errors_effector.rs

```diff
@@ -14,9 +14,10 @@
  * limitations under the License.
  */
 
-/// This trait is intended to figuring out whether a last error should be set or not.
-pub(crate) trait LastErrorAffectable {
+/// This trait controls whether to set %last_error% and :error: or not.
+pub(crate) trait InstructionErrorsEffector {
```

Comment:
InstructionErrorsEffector is an entity that effects instruction errors. ErrorAffectable is something that gets an affect from an error.

Reply:
Let's check their meaning [here](https://www.touro.edu/departments/writing-center/tutorials/affect-or-effect/):

> Affect is almost always used as a verb to mean to influence someone or something, rather than to cause something. 

> Effect is most often used as a noun. It points toward an event or a thing. It is often used when an end result is being discussed.

> Here are some synonyms of affect: alter, change, influence, modify and impact (the verb version). That list should affect your understanding of the word. In this case, affect would mean “improve.”

> Some synonyms of effect include words like result, repercussion, consequence, outcome, aftermath and the noun version of impact.

So, affect is not about negative effect, and here it clearly means that an instruction _alter, change, influence, modify and impact_ an error. So affect is a natural choice here, although there is the second point of view about Marine.

Reply:
Agree.